src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/X86/Avx2.PlatformNotSupported.cs

```diff
@@ -13,13 +13,13 @@ namespace System.Runtime.Intrinsics.X86
     [CLSCompliant(false)]
     public abstract class Avx2 : Avx
     {
-        internal Avx2() { }
+        protected internal Avx2() { }
```

Comment:
We'll need to be careful here to ensure this doesn't get exposed in the public surface area.

It would be nice if we could get this working while keeping `AvxVnni` in System.Runtime.Intrinsics.Experimental.